modules/mauth-authenticator-http4s/src/test/scala/com/mdsol/mauth/http4s/MAuthMiddlewareSuite.scala

```diff
@@ -0,0 +1,268 @@
+
+  private val service = MAuthMiddleware.httpRoutes[IO](requestValidationTimeout)(route).orNotFound
+
+  val authenticatorV2: RequestAuthenticator = new RequestAuthenticator(client, epochTimeProvider, v2OnlyAuthenticate = true)
+  val serviceV2 =
+    MAuthMiddleware.httpRoutes[IO](requestValidationTimeout)(route)(implicitly[Async[IO]], authenticatorV2, implicitly[ExecutionContext]).orNotFound
+
+  test("allow successfully authenticated request") {
+    val res = service(
+      Request[IO](GET, uri"/").withHeaders(
+        MAuthRequest.X_MWS_TIME_HEADER_NAME -> timeHeader.toString,
+        MAuthRequest.X_MWS_AUTHENTICATION_HEADER_NAME -> authHeader
+      )
+    )
+
+    res.map(_.status).assertEquals(Status.Ok)
+  }
+
+  test("allow successfully authenticated request with both v1 and v2 headers, with V2 headers taking precedence") {
+    val res = service(
+      Request[IO](GET, uri"/").withHeaders(
+        MAuthRequest.MCC_TIME_HEADER_NAME -> timeHeader.toString,
+        MAuthRequest.MCC_AUTHENTICATION_HEADER_NAME -> authHeaderV2,
+        MAuthRequest.X_MWS_TIME_HEADER_NAME -> (timeHeader - requestValidationTimeout.toSeconds - 10).toString,
+        MAuthRequest.X_MWS_AUTHENTICATION_HEADER_NAME -> "invalid auth header"
+      )
+    )
+
+    res.map(_.status).assertEquals(Status.Ok)
+  }
+
+  test("allow successfully authenticated request with both v1 and v2 headers, fallback to v1 if v2 failed") {
+    val res = service(
+      Request[IO](GET, uri"/").withHeaders(
+        MAuthRequest.MCC_TIME_HEADER_NAME -> timeHeader.toString,
+        MAuthRequest.MCC_AUTHENTICATION_HEADER_NAME -> wrongAuthHeaderV2,
+        MAuthRequest.X_MWS_TIME_HEADER_NAME -> timeHeader.toString,
+        MAuthRequest.X_MWS_AUTHENTICATION_HEADER_NAME -> authHeader
+      )
+    )
+
+    res.map(_.status).assertEquals(Status.Ok)
+  }
+
+  test("reject request if validation times out") {
+    val res = service(
+      Request[IO](GET, uri"/").withHeaders(
+        MAuthRequest.X_MWS_TIME_HEADER_NAME -> (timeHeader - requestValidationTimeout.toSeconds - 10).toString,
+        MAuthRequest.X_MWS_AUTHENTICATION_HEADER_NAME -> authHeader
+      )
+    )
+
+    res
+      .redeem(
+        {
+          case e if e.isInstanceOf[MAuthValidationException] => true // TODO: fix removing the throwing of random Exceptions
+          case _                                             => false // any other exception shouldn't be thrown
+        },
+        _ => false
+      )
+      .assert
+  }
+
+  test("reject if public key cannot be found") {
+    val localClient = new ClientPublicKeyProvider {
+      override def getPublicKey(appUUID: UUID): Future[Option[PublicKey]] =
+        if (appUUID == appUuid) {
+          Future(none)
+        } else Future.failed(new Throwable("Wrong app UUID"))
+      override def getPublicKeyIO(appUUID: UUID): IO[Option[PublicKey]] = ???
```

Comment:
Can you remove this from the public interface while you're at it? :) I don't think it's suppose to be public

Reply:
Yep, on it!

Reply:
I think we did it on purpose so you dont need to use future in places like this. The future API is just calling this with an .unsafeToFuture.

Reply:
If its not used anywhere though its fine to remove it.